forum/views/search.py

```diff
@@ -30,19 +31,67 @@ class SearchThreadsView(APIView):
 
     permission_classes = (AllowAny,)
 
-    def _validate_request(self, request: Request) -> None:
+    def _validate_and_extract_params(self, request: Request) -> dict[str, Any]:
         """
-        Validate query params in the request.
+        Validate and extract query parameters from the request.
         """
-        course_id = request.GET.get("course_id")
+        params: dict[str, Any] = {}
+
+        # Required parameters
         text = request.GET.get("text")
-        sort_key = request.GET.get("sort_key")
-        if not course_id:
-            raise ValueError("course_id is required")
         if not text:
             raise ValueError("text is required")
-        if sort_key and sort_key not in ["activity", "comments", "date", "votes"]:
+        params["text"] = text
+
+        # Sort key validation
+        sort_key = request.GET.get("sort_key")
+        if not sort_key or sort_key == "None":
+            sort_key = "date"
+        elif sort_key and sort_key not in ["activity", "comments", "date", "votes"]:
             raise ValueError("invalid sort_key")
+        params["sort_key"] = sort_key or None
+
+        # Pagination handling
+        try:
+            if request.GET.get("page") in ["None", None]:
+                params["page"] = settings.FORUM_DEFAULT_PAGE
+            else:
+                params["page"] = int(
+                    request.GET.get("page", settings.FORUM_DEFAULT_PAGE)
+                )
+        except ValueError as exc:
+            raise ValueError("Invalid page value.") from exc
```

Comment:
only `else` clause will raise exception so we could write it like this:
```
page = request.GET.get("page")

if page in [None, "None"]:
    params["page"] = settings.FORUM_DEFAULT_PAGE
else:
    try:
        params["page"] = int(page)
    except ValueError as exc:
        raise ValueError("Invalid page value.") from exc
```

Similarly, we can update for `per_page` param.